dev/benchmarks/platform_views_layout/lib/android_platform_view.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+
+class AndroidPlatformView extends StatelessWidget {
+  /// Creates a platform view for Android, which is rendered as a
+  /// native view.
+  /// `viewType` identifies the type of Android view to create.
+  const AndroidPlatformView({
+    Key key,
+    @required this.viewType,
+  }) : assert(viewType != null),
+       super(key: key);
+
+  /// The unique identifier for the view type to be embedded by this widget.
+  ///
+  /// A PlatformViewFactory for this type must have been registered.
+  final String viewType;
+
+  @override
+  Widget build(BuildContext context) {
+    return PlatformViewLink(
+      viewType: viewType,
+      onCreatePlatformView: _onCreateAndroidView,
+      surfaceFactory: (BuildContext context, PlatformViewController controller) {
+        return PlatformViewSurface(
+          controller: controller,
+          gestureRecognizers: const <Factory<OneSequenceGestureRecognizer>>{},
+          hitTestBehavior: PlatformViewHitTestBehavior.opaque,
+        );
+      },
+    );
+  }
+
+  PlatformViewController _onCreateAndroidView(PlatformViewCreationParams params) {
+    final _AndroidViewController controller = _AndroidViewController(params.id, viewType);
+    controller._initialize().then((_) { params.onPlatformViewCreated(params.id); });
+    return controller;
+  }
+}
+
+// TODO(egarciad): The Android view controller should be defined in the framework.
```

Comment:
What is the motivation of this change? (I mean add AndroidViewController to the framework)
Also, please link an issue for the TODOs, this one and the below ones.

Reply:
I added a link to https://github.com/flutter/flutter/issues/55904. The other TODOs don't need an individual issue since they are all related to the same issue.